vision/product_search/create_product.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+// [START imports]
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	
+	"context"
+
+	vision "cloud.google.com/go/vision/apiv1"
+	visionpb "google.golang.org/genproto/googleapis/cloud/vision/v1"
+)
+// [END imports]
+
+// [START vision_product_search_create_product]
+func createProduct(w io.Writer, projectId string, location string, productId string, productDisplayName string, productCategory string) error {
```

Comment:
Do all of the languages have this many parameters? In general, we try to have as few parameters as possible.

Reply:
it is so: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/vision/cloud-client/product_search/product_management.py#L41

if there are more go-idiomatic ways of dealing with this, please let me know.